### ISSUE TITLE
AWS: Feature Request : Can you create an "ADD" action for security group resource so that we can add a default rule for all security groups

### ISSUE THREAD

**Issue by sri321** [cloud-custodian/cloud-custodian#3323]

**Comment by tohch4** [comment]
@kapilt per discussion in gitter, I see this appears to be related to cloud-custodian#2622 and cloud-custodian#3220. Is this implemented and I misunderstood the docs and forthcoming PRs, or this is not currently possible?

**Comment by kapilt** [comment]
@tohch4 thats correct, no one has implemented the requested feature, contributions welcome.

**Comment by tohch4** [comment]
@kapilt, sounds like an exciting opportunity and a more pressing need for me. Thanks for the encouragement.

This is one of the more elaborate Python projects I have used, so I presume I need [to start with a function in the AWS VPC resources file and do the opposite of remove-permissions](https://github.com/cloud-custodian/cloud-custodian/blob/e5999303b7d3af31524109fbaf4fd2af7a920f94/c7n/resources/vpc.py#L1216)? Additionally, not 100% obvious to me, but a safe assumption, I assume the `actions:` action items execute in order, such that if I do this:

```
- remove-permissions
- add-permissions
  - new-property
```

It executes `remove-permissions` first, then `add-permissions`?

**Comment by tohch4** [comment]
I am going to drop this as I was too slow to beat out ajmisra in #4973 and that PR looks very promising, for those potentially following this issue only.
